activiti-cloud-services-runtime-bundle/messages-events/src/main/java/org/activiti/cloud/services/messages/events/producer/StartMessageDeployedEventMessageProducer.java

```diff
@@ -35,9 +36,20 @@ public void onEvent(StartMessageDeployedEvent event) {
             throw new IllegalStateException("requires active transaction synchronization");
         }
         
-        Message<CloudStartMessageDeployedEvent> message = messageBuilderFactory.create(event)
-                                                                               .withPayload(runtimeEventConverter.from(event))
-                                                                               .build();
+        StartMessageSubscription messageSubscription = event.getEntity()
+                                                            .getMessageSubscription();
+
+        MessageEventPayload messageEventPayload = MessageEventPayloadBuilder.messageEvent(messageSubscription.getEventName())
+                                                                            .withCorrelationKey(messageSubscription.getConfiguration())
+                                                                            .build();
+        
+        
+        Message<MessageEventPayload> message = messageBuilderFactory.create(event)
```

Comment:
The content of the different message producer is very similar, especially `BpmnMessageReceivedEventMessageProducer`, `BpmnMessageSentEventMessageProducer` and `BpmnMessageWaitingEventMessageProducer`. Would it make sense to extract common code into a separated class?

Reply:
I think it's a good practice to use explicit bpmn listeners to consume engine message events. But I agree that the repeating  boilerplate code could be refactored.

Reply:
Fixed in 08ab8b8953897644f446f0eb5d6e759115bfc43f